### DISCUSSION TITLE
How to update without losing data?

### DISCUSSION THREAD

**Discussion by Equity7** [plan-player-analytics/Plan#2394] [Q&A] [answered]
I### Describe the issue
I couldn't find anywhere else to post a simple question. How do you run updates without losing all prior data?

**Comment by AuroraLS3** [comment]
Replace the jar(s) and restart the server(s)

**Comment by Equity7** [comment]
I lost all my data I'm afraid.
1)So was my mistake in deleting the prior folder as well as the prior .jar? I should leave the folder?
2)I just redo the config file each time there is an update and that is all?

**Comment by AuroraLS3** [comment]
Yes don't delete the Plan folder, if you're using sqlite that contains the database

Config file doesn't need to be touched, incompatible parts in there are automatically updated.

**Comment by Equity7** [comment]
Oh ok so I don't have to reenter MySql data. Thanks so much it is probably the best of the 30+ plugins I use!

**Comment by Equity7** [comment]
Actually may I ask one more question? I have another server and I have this plugin on it separately. It is not Bungee cord. Is there a way I can put both servers on the one rollup? From what I have read it might be possible?

**Comment by AuroraLS3** [comment, marked as answer]
Now there is a wiki page https://github.com/plan-player-analytics/Plan/wiki/How-to-update-the-plugin
